### ISSUE TITLE
External CSS and JS files not loading

### ISSUE THREAD

**Issue by wahabkhan1892** [me-no-dev/ESPAsyncWebServer#256]
I have an index.html file with links to several JS and CSS files, a couple of which are 54KB and 300KB. I am using ESP8266 in AP mode. When I connect to it and type in 192.168.4.1, I can see through Wireshark capture, that index.html is being loaded properly and after that, the client requests 5 new ports. The server acknowledges these requests. The client then sends GET requests for the JS and CSS files on these ports, which are also acknowledged by the server. But after that the server doesn't send anything, when its supposed to start sending data packets.
![capt_0](https://user-images.githubusercontent.com/32616242/31445090-a691dc88-ae9d-11e7-8680-091d7aa93750.PNG)
![capt_1](https://user-images.githubusercontent.com/32616242/31445091-a6ce83ae-ae9d-11e7-8855-89ada6c61fee.PNG)

**Comment by jeremypoulter** [comment]
This sounds very much like issue #194 I raised a while ago. How much memory do you have free? FYI, I ended up working around this issue by a combination of embedding all the resources in PROGMEM (and avoiding SPIFFS as that allocates a large chunk of memory), combining all JS/CSS files into a single file and only using a small buffer when reading from flash and sending on to AsyncWebServer, see https://github.com/OpenEVSE/ESP8266_WiFi_v2.x

**Comment by wahabkhan1892** [comment]
Is there a way to reduce memory usage while keeping SPIFFS and all the files?

**Comment by rick-bp** [comment]
I have the same problem.
#245

**Comment by bah-sa** [comment]
I have the same problem.
Only one linked file in page (CSS) and it do not loading 8(

**Comment by me-no-dev** [comment]
So all of you have this issue only in AP mode?

**Comment by wahabkhan1892** [comment]
Apparently yes, only in AP mode. I managed to get around this by packing all external files into one index file. In that way, the client doesn't have to open extra sockets, and the device doesn't run out of memory.

**Comment by bah-sa** [comment]
I have problem in STA mode.
But I not sure it is AsyncServer issue.
Linked stylesheet in page dont applied on any mobile browser (chrome, samsung, opera mini), but on PC MSIE successfully applied.
Logs show web server send file, but mobile client browser dont apply it.
Postman succesfully load it also, content-type in headers is correct 'text/css'.
8(

**Comment by bah-sa** [comment]
Problem fixed. 8)
CSS-file must be in UTF-8 encoding with BOM (!!!), otherwise stylesheet not applied.

**Comment by thomasgresch** [comment]
Had the same problem - combing the files worked for me to, but I would be interested in a more elegant solution.

**Comment by vindolin** [comment]
@bah-sa strange, that doesn't seem to make sense.
Can you copy the request HAR with Chrome or Firefox and post it here?

**Comment by briangraphted** [comment]
I had this problem too. My esp was loading index.htm, style.css, and 3 separate .js files. I spent time minifying all the files but that didn’t solve the issue. Then I combined two of the .js files so I had 4 downloads total (index + css + js+ js). Then it worked!  I believe the key here is in the (great) analysis by wahabhkah1892 mentioned above “the client requests 5 new ports”. The esp either can’t handle that many (hard coded connection pool?) or runs out of memory or both. My observed rule of thumb: limit your files to 4 (and do all the good hygiene of minimizing them, and if needed compress and embed them [in that order]). Good luck!
